src/lib/comparison.ts

```diff
@@ -98,19 +131,62 @@ export class Comparison<Location> {
     const matchingFiles: Matches<Location> = new Map();
     const [ast, mapping] = await this.tokenizer.tokenizeFileWithMapping(file);
     for await (const { hash, location } of hashFilter.hashesFromString(ast)) {
+      if (this.filteredHashSet.has(hash)) {
+        continue;
+      }
       const matches = this.index.get(hash);
-      if (matches) {
-        for (const [fileName, lineNumber] of matches) {
-          const match: [Location, Location] = [lineNumber, mapping[location]];
-          const lines = matchingFiles.get(fileName);
-          if (lines) {
-            lines.push(match);
-          } else {
-            matchingFiles.set(fileName, [match]);
-          }
+      if (!matches || !this.filterByFragmentCount(matches.length)) {
+        continue;
+      }
+
+      for (const [fileName, lineNumber] of matches) {
+        // add the match if the match is not with the file and if the file comes first when alphabetically sorted.
+        // This is done to avoid the duplicates in the following case: when file A matches with file B, file B will
+        // also match with file A.
+        if (fileName === file) {
+          continue;
+        }
+        const match: [Location, Location] = [lineNumber, mapping[location]];
+        const lines = matchingFiles.get(fileName);
+        if (lines) {
+          lines.push(match);
+        } else {
+          matchingFiles.set(fileName, [match]);
         }
       }
     }
     return matchingFiles;
   }
+
+  /**
+   * Add a file to the filter index.
+   * @param file The file name of the file you want to add.
+   */
+  public async addFileToFilterList(file: string): Promise<void> {
+    try {
+      const [ast] = await this.tokenizer.tokenizeFileWithMapping(file);
+      for await (const { hash } of this.noFilter.hashesFromString(ast)) {
+        this.filteredHashSet.add(hash);
+      }
+    } catch (error) {
+      console.error(`There was a problem parsing ${file}. ${error}`);
+      return; // this makes sure the promise resolves instead of rejects
+    }
+  }
+
+  /**
+   *
+   * @param fragmentCount The amount of times a certain code fragment has appeared
+   * @returns true if is value is acceptable under the options given in the contructor or when the options are not
+   * defined.
+   */
+  private filterByFragmentCount(fragmentCount: number): boolean {
```

Comment:
This naming can be confusing. It used to be a fragment count filter, but right now it is used as a hash count filter. (The name fragment is used as a synonym for range to the end-user)